pkg/registry/core/shoot/strategy.go

```diff
@@ -190,13 +189,6 @@ func cleanupAdmissionPlugins(shoot *core.Shoot) {
 	shoot.Spec.Kubernetes.KubeAPIServer.AdmissionPlugins = admissionPlugins
 }
 
-// dropEnableBasicAuthenticationField sets the enableBasicAuthentication to nil.
-func dropEnableBasicAuthenticationField(shoot *core.Shoot) {
-	if shoot.Spec.Kubernetes.KubeAPIServer != nil {
-		shoot.Spec.Kubernetes.KubeAPIServer.EnableBasicAuthentication = nil
-	}
-}
-
 // setKubernetesDashboardAuthMode sets the kubernetes-dashboard authentication mode to "token", if its current value is not "token" (for example "basic").
 func setKubernetesDashboardAuthMode(shoot *core.Shoot) {
```

Comment:
Having a second look, let's also drop the `setKubernetesDashboardAuthMode` func. It was a migration code that did its job.